lib/queue.rb

```diff
@@ -1,31 +1,59 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(30)
+    @front = @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+    
+    if !@store[@back].nil? && @front == @back
+      raise StandardError.new('queue is full.')
+    end
+    
+    @store[@back] = element
+    @back = (@back+1) % @store.length
+    
   end
-
+  
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    dequeued = @store[@front]
+    
+    # remove dequeued item from queue
+    @store[@front] = nil
+    # move front to next item in queue
+    @front = @front+1
```

Comment:
You need to make sure the front will wrap around the queue

```suggestion
    @front = (@front+1) % @store.length
```

You also need to check to see if the queue is empty!